src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/representer/CommonRepresenter.kt

```diff
@@ -74,16 +75,7 @@ open class CommonRepresenter(
 
     /** Create [Node] for [Byte], [Short], [Int], [Long], [Float], [Double] */
     protected val representNumber = RepresentToNode { data: Any ->
-        // For JS, some numbers (known so far: floats, infinity, -infinity and NaN)
-        // are incorrectly identified as the below types.
-        // Because of it, they are represented differently than for other platforms.
-        // TODO: fix within https://github.com/krzema12/snakeyaml-engine-kmp/issues/526
-        if (
-            data is Byte
-            || data is Short
-            || data is Int
-            || data is Long
-        ) {
+        if (isInteger(data as Number)) {
```

Comment:
suggesion: we can move cast to Number outside of if statement because it is used in both branches

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/579/commits/e3e237f9776e9bb53aacf5f0c9a4c6f3338a3ffe